src/pam_unix_ng-common.c

```diff
@@ -25,20 +25,37 @@ parse_args(pam_handle_t *pamh, int flags, int argc, const char **argv,
   cfg->fail_delay = 2000;
   cfg->minlen = 8;
 
+  /*
+   * strictly spoken this reading of the logindefs has nothing to do with
+   * parsing PAM module arguments. it is rather a kind of default
+   * initialization from system file settings.
+   *
+   * I also see only one user of `init_crypt == true` in `pam_sm_chauthtok()`.
+   *
+   * I believe it would make more sense to move this into a dedicated function
+   * which is only called in `pam_sm_chauthtok()`.
+   */
   /* only read login.defs if we change the password */
   if (init_crypt)
     {
-      _cleanup_free_ char *val;
+      char *val;
 
       cfg->crypt_count = 0;
+      cfg->crypt_prefix = "$y$";
 
       val = get_logindefs_string("ENCRYPT_METHOD", NULL);
       if (isempty(val) || streq(val, "YESCRYPT"))
+	// this string is already assigned above.
+	// maybe this default value can be placed in an `else` branch for all
+	// three cases dealt with here.
```

Comment:
Yes, the string is the default, but if you change the default you don't need to remember to implement YESCRYPT here.
But moved to "else".